bin/liquidator/src/tx_sender.rs

```diff
@@ -0,0 +1,130 @@
+use crate::liquidation_state::LiquidationState;
+use crate::tcs_state::TcsState;
+use crate::SharedState;
+use mango_v4_client::AsyncChannelSendUnlessFull;
+use std::sync::{Arc, RwLock};
+use tokio::task::JoinHandle;
+use tracing::debug;
+
+pub fn spawn_tx_senders_job(
+    max_parallel_operations: u64,
+    tx_trigger_receiver: async_channel::Receiver<()>,
+    rebalance_trigger_sender: async_channel::Sender<()>,
+    shared_state: Arc<RwLock<SharedState>>,
+    liquidation: Box<LiquidationState>,
+    tcs: Box<TcsState>,
+) -> Vec<JoinHandle<()>> {
+    if max_parallel_operations < 1 {
+        tracing::error!("max_parallel_operations must be >= 1");
+        std::process::exit(1)
+    }
+
+    let workers: Vec<JoinHandle<()>> = (0..max_parallel_operations)
+        .map(|i| {
+            tokio::spawn({
+                let shared_state = shared_state.clone();
+                let receiver = tx_trigger_receiver.clone();
+                let rebalance_trigger_sender = rebalance_trigger_sender.clone();
+                let mut liquidation = liquidation.clone();
+                let mut tcs = tcs.clone();
+                let id = i;
+                async move {
+                    loop {
+                        debug!("Worker #{} waiting for task", id);
+                        let _ = receiver.recv().await.unwrap();
+
+                        // a task is supposed to be available to process
+                        // find it in global shared state, and mark it as processing
+                        // (it's also possible we don't find anything in case liq/tcs job enqueued the same task multiple time)
+                        let (l, t) = {
+                            let mut writer = shared_state.write().unwrap();
+
+                            // print out list of all task for debugging
+                            for x in &writer.liquidation_candidates_accounts {
+                                if !writer.processing_liquidation.contains(x) {
+                                    tracing::trace!(" - LIQ {:?}", x);
```

Comment:
minor, I often like to bring the info!/trace!/ etc macros into file scope; also you could write "{x:?}"